daml-foundations/daml-tools/da-hs-daml-cli/DA/Cli/Damlc.hs

```diff
@@ -225,7 +225,8 @@ execIde telemetry (Debug debug) = NS.withSocketsDo $ do
     opts <- liftIO $ defaultOptionsIO Nothing
     withLogger $ \loggerH ->
         withScenarioService loggerH $ \scenarioService -> do
-            execInit (ProjectOpts Nothing (ProjectCheck "" False)) (InitPkgDb True)
+            -- TODO we should allow different LF versions in the IDE.
```

Comment:
Do you mean if we should be able to pass `--target` when running the IDE?

Reply:
That would be a first step but given that users don’t start the IDE process manually, it doesn’t help much. We probably want to consider adding an option in `daml.yaml` for the target LF versions so that this can be picked up automatically and all commands agree on the LF version.

Reply:
Or we only support one LF version, and say that if you want to use an old LF version, we demand you use an old SDK. I'm not inclined to complicate everything to allow people to downgrade just one aspect of the tool. The one exception is perhaps --lf=dev, to test the next version, but having that not work in the IDE is no big deal.

Reply:
Making it a hidden flag and only supporting `dev` would be fine with me as well. What I don’t like is having it be a user-facing option (and given that this PR was the result of a bug report, one that users do use) that is not supported when using the IDE.

Reply:
Agreed - the current state is not right. Simplifying the user experience seems better, since I think telling the user to downgrade would have been fine.